src/components/editor/selections.js

```diff
@@ -2,35 +2,40 @@
 
 const markExtend = (selection, markType) => {
   if (!selection) return
-  const { $cursor, $anchor } = selection
-  const pos = $cursor || $anchor
-  let startIndex = pos.index()
-  let endIndex = pos.indexAfter()
-
-  const hasMark = (index) => {
-    // clicked outside edge of tag.
-    if (index === pos.parent.childCount) {
-      index--
+
+  const hasMark = (pos, index) =>
```

Comment:
Let's move the helper functions out of `markExtend` -- they don't really require the closure state (everything they need can be passed as arguments) so there's no point in re-creating them in every invocation.